### PR TITLE
Add a linter for calls to omp_set_num_threads

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #6272

Probably, the step to pre-process the file to strip comments will be repeated by other linters, perhaps we should _just_ work from the preprocessed file for the C linters?

Leaving the simple version for now that's more parallel to the existing code.

### CODE REVIEW

**Comment by MichaelChirico** [comment]
* **#6347** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/6347?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6346** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/6346?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#6345** <a href="https://app.graphite.dev/github/pr/Rdatatable/data.table/6345?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MichaelChirico and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: .ci/linters/c/omp_set_num_threads_linter.R [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+# Ensure no calls to omp_set_num_threads() [to avoid affecting other packages and base R]
+# Only comments referring to it should be in openmp-utils.c
+omp_set_num_threads_linter = function(c_file) {
+  # strip comments, we only care if the function appears in actual code.
+  processed_lines = system2("gcc", c("-fpreprocessed", "-E", "-xc", "-", c_file), stdout=TRUE, stderr=FALSE)
```
Comment (ben-schwen) [inline .ci/linters/c/omp_set_num_threads_linter.R:6]:
```suggestion
  processed_lines = system2("gcc", c("-fpreprocessed", "-E", "-xc", c_file), stdout=TRUE, stderr=FALSE)
```
I guess the empty option is a typo?

Comment (MichaelChirico) [reply .ci/linters/c/omp_set_num_threads_linter.R]: Ah, that is for using `stdin=c_file`, I was trying out `stdin=lines`, which didn't work, `-xc -` is leftover from that. It's easiest to just use the filename directly.

**Review by ben-schwen — APPROVED** [review]
LGTM

**Pull request merged**